packages/pwa/app/utils/url.js

```diff
@@ -78,28 +80,45 @@ export const buildUrlSet = (url = '', key = '', values = [], extraParams = {}) =
  * Given a category and the current locale returns an href to the product list page.
  *
  * @param {Object} category
- * @param {string} locale
+ * @param {Object} options
  * @returns {string}
  */
-export const categoryUrlBuilder = (category, locale = DEFAULT_LOCALE) =>
-    encodeURI(`/${locale}/category/${category.id}`)
+export const categoryUrlBuilder = (category, options) => {
+    const {locale = DEFAULT_LOCALE} = options
+    console.log('locale', locale)
+    const url = `/category/${category.id}`
+    const updatedUrl = buildPathWithLocaleAndSiteParams(url, {locale})
+    console.log('updatedUrl', updatedUrl)
+    return encodeURI(updatedUrl)
+}
 
 /**
  * Given a product and the current locale returns an href to the product detail page.
  *
  * @param {Object} product
- * @param {string} locale
+ * @param {Object} options
  * @returns {string}
  */
-export const productUrlBuilder = (product) => encodeURI(`/product/${product.id}`)
+export const productUrlBuilder = (product, options = {}) => {
```

Comment:
Open thought: 

Url modification happens in 2 places, from _basename_ and `buildPathWithLocaleAndSiteParams`, both happening in different places. I always thought that was less ideal. Would we be able to move `buildPathWithLocaleAndSiteParams` usage into our Link component? Feels like it's a little closer to where _basename_ gets applied. This will probably break usages of history.push but those can be fixed. This idea is half bakes, but maybe we can think about it.

Reply:
Thanks for the comment, the Link component has already used the `buildPathWithLocaleAndSiteParams`.

I just had another look at some components (nest-accodion and list-menu), it turned out that these component is not using our `Link` component. They are still using the `RouterLink` from react-router-dom. Hence, I've made the change to use our Link component. This mean we don't have to modify our urlBuilder util funcs with `buildPathWithLocaleAndSiteParams`, the Link component should handle the logic within itself